<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="mistyR">
<title>Functional analysis with MISTy - pathway activity and ligand expression • mistyR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Functional analysis with MISTy - pathway activity and ligand expression">
<meta property="og:description" content="mistyR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-9"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-9');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mistyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.10.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/mistyR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/FunctionalAndStructuralPipeline.html">Learning functional and structural spatial relationships with MISTy</a>
    <a class="dropdown-item" href="../articles/FunctionalPipelinePathwayActivityLigands.html">Functional analysis with MISTy - pathway activity and ligand expression</a>
    <a class="dropdown-item" href="../articles/FunctionalPipelinePathwaySpecific.html">Functional analysis with MISTy - pathway specific genes</a>
    <a class="dropdown-item" href="../articles/MistyRStructuralAnalysisPipelineC2L.html">Structural analysis with MISTy - based on cell2location deconvolution</a>
    <a class="dropdown-item" href="../articles/MistyRStructuralAnalysisPipelineDOT.html">Structural analysis with MISTy - based on DOT deconvolution</a>
    <a class="dropdown-item" href="../articles/mistyDataFormats.html">mistyR and data formats</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/mistyR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://saezlab.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Functional analysis with MISTy - pathway activity and ligand expression</h1>
                        <h4 data-toc-skip class="author">Leoni
Zimmermann</h4>
            <address class="author_afil">
      Heidelberg University, Heidelberg,
Germany<br><h4 data-toc-skip class="author">Jovan
Tanevski</h4>
            <address class="author_afil">
      Heidelberg University and Heidelberg University Hospital,
Heidelberg, Germany <br>Jožef Stefan Institute, Ljubljana,
Slovenia<br><a class="author_email" href="mailto:#"></a><a href="mailto:jovan.tanevski@uni-heidelberg.de" class="email">jovan.tanevski@uni-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2024-03-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/mistyR/blob/HEAD/vignettes/FunctionalPipelinePathwayActivityLigands.Rmd" class="external-link"><code>vignettes/FunctionalPipelinePathwayActivityLigands.Rmd</code></a></small>
      <div class="d-none name"><code>FunctionalPipelinePathwayActivityLigands.Rmd</code></div>
    </address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>10X Visium captures spatially resolved transcriptomic profiles in
spots containing multiple cells. In this vignette, we will use the gene
expression information from Visium data to infer pathway activity and
investigate spatial relationships between some pathways and ligand
expression.</p>
<p>Load the necessary packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># MISTy </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/mistyR/">mistyR</a></span><span class="op">)</span> </span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://future.futureverse.org" class="external-link">future</a></span><span class="op">)</span> </span>
<span></span>
<span><span class="co">#Seurat </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mojaveazure.github.io/seurat-object/" class="external-link">SeuratObject</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Data manipulation </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Pathways and annotation</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/decoupleR/" class="external-link">decoupleR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r.omnipathdb.org/" class="external-link">OmnipathR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/saezlab/progeny" class="external-link">progeny</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Cleaning names</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sfirke/janitor" class="external-link">janitor</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-and-load-data">Get and load data<a class="anchor" aria-label="anchor" href="#get-and-load-data"></a>
</h2>
<p>For this showcase, we use a 10X Visium spatial slide from <a href="https://doi.org/10.1038/s41586-022-05060-x" class="external-link">Kuppe et al.,
2022</a>, where they created a spatial multi-omic map of human
myocardial infarction. The tissue example data comes from the human
heart of patient 14, which is in a later state after myocardial
infarction. The Seurat object contains, among other things, the
normalized and raw gene counts. First, we have to download and extract
the file:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/6580069/files/10X_Visium_ACH005.tar.gz?download=1"</span>,</span>
<span>    destfile <span class="op">=</span> <span class="st">"10X_Visium_ACH005.tar.gz"</span>, method <span class="op">=</span> <span class="st">"curl"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="st">"10X_Visium_ACH005.tar.gz"</span><span class="op">)</span></span></code></pre></div>
<p>The next step is to load the data, extract the normalized gene
counts, names of genes expressed in at least 5% of the spots, and pixel
coordinates. It is recommended to use pixel coordinates instead of row
and column numbers since the rows are shifted and therefore do not
express the real distance between the spots.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_vs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"ACH005/ACH005.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">seurat_vs</span>, layer <span class="op">=</span> <span class="st">"counts"</span>, assay <span class="op">=</span> <span class="st">"SCT"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">gene_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">expression</span><span class="op">[</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">expression</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">0.05</span>,<span class="op">]</span><span class="op">)</span> </span>
<span><span class="va">geometry</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/GetTissueCoordinates.html" class="external-link">GetTissueCoordinates</a></span><span class="op">(</span><span class="va">seurat_vs</span>, scale <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pathway-activity">Pathway activity<a class="anchor" aria-label="anchor" href="#pathway-activity"></a>
</h2>
<p>Now we create a Seurat object with pathway activities inferred from
<a href="https://saezlab.github.io/progeny/index.html" class="external-link"><code>PROGENy</code></a>.
We delete the PROGENy assay done by Kuppe et al. and load a model matrix
with the top 1000 significant genes for each of the 15 available
pathways. We then extract the genes that are both common to the PROGENy
model and the snRNA-seq assay from the Seurat object. We compute the
weighted sum of both and scale them to infer the pathway activity. We
save the result in a Seurat assay and clean the row names to handle
problematic variables.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_vs</span><span class="op">[[</span><span class="st">'progeny'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="co"># Matrix with important genes for each pathway</span></span>
<span><span class="va">model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/decoupleR/reference/get_progeny.html" class="external-link">get_progeny</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="st">"human"</span>, top <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use multivariate linear model to estimate activity</span></span>
<span><span class="va">est_path_act</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/decoupleR/reference/run_mlm.html" class="external-link">run_mlm</a></span><span class="op">(</span><span class="va">expression</span>, <span class="va">model</span>,.mor <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Put estimated pathway activities object into the correct format</span></span>
<span><span class="va">est_path_act_wide</span> <span class="op">&lt;-</span> <span class="va">est_path_act</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>id_cols <span class="op">=</span> <span class="va">condition</span>, names_from <span class="op">=</span> <span class="va">source</span>, values_from <span class="op">=</span> <span class="va">score</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"condition"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Clean names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">est_path_act_wide</span><span class="op">)</span>  <span class="op">&lt;-</span> <span class="va">est_path_act_wide</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://sfirke.github.io/janitor/reference/clean_names.html" class="external-link">clean_names</a></span><span class="op">(</span>parsing_option <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a Seurat object</span></span>
<span><span class="va">seurat_vs</span><span class="op">[[</span><span class="st">'progeny'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">est_path_act_wide</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Format for running MISTy later</span></span>
<span><span class="va">pathway_activity</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">seurat_vs</span>, <span class="st">"progeny"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="ligands">Ligands<a class="anchor" aria-label="anchor" href="#ligands"></a>
</h2>
<p>To annotate the expressed ligands found in the tissue slide, we
import an intercellular network of ligands and receptors from <a href="https://omnipathdb.org/" class="external-link">Omnipath</a>. We extract the ligands that
are expressed in the tissue slide and get their count data. We again
clean the row names to handle problematic variables.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get ligands</span></span>
<span><span class="va">lig_rec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/OmnipathR/reference/import_intercell_network.html" class="external-link">import_intercell_network</a></span><span class="op">(</span>interactions_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>datasets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ligrecextra'</span>, <span class="st">'omnipath'</span>, <span class="st">'pathwayextra'</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                         transmitter_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>parent <span class="op">=</span> <span class="st">'ligand'</span><span class="op">)</span>,</span>
<span>                         receiver_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>parent <span class="op">=</span> <span class="st">'receptor'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get unique ligands</span></span>
<span><span class="va">ligands</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">lig_rec</span><span class="op">$</span><span class="va">source_genesymbol</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get expression of ligands in slide</span></span>
<span><span class="va">slide_markers</span> <span class="op">&lt;-</span> <span class="va">ligands</span><span class="op">[</span><span class="va">ligands</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">gene_names</span><span class="op">]</span> </span>
<span><span class="va">ligand_expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">expression</span><span class="op">[</span><span class="va">slide_markers</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://sfirke.github.io/janitor/reference/clean_names.html" class="external-link">clean_names</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#clean names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seurat_vs</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">SCT</span><span class="op">@</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">seurat_vs</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">SCT</span><span class="op">@</span><span class="va">data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://sfirke.github.io/janitor/reference/clean_names.html" class="external-link">clean_names</a></span><span class="op">(</span>parsing_option <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-pathway-activity">Visualize pathway activity<a class="anchor" aria-label="anchor" href="#visualize-pathway-activity"></a>
</h2>
<p>Before continuing with creating the MISTy view, we can look at the
slide itself and some of the pathway activities.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Slide</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialPlot</a></span><span class="op">(</span><span class="va">seurat_vs</span>, alpha <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pathway activity examples</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_vs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"progeny"</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">seurat_vs</span>, feature <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mapk"</span>, <span class="st">"p53"</span><span class="op">)</span>, keep.scale <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-6-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="misty-views">MISTy views<a class="anchor" aria-label="anchor" href="#misty-views"></a>
</h2>
<p>Now we need to create the MISTy views of interest. We are interested
in the relationship of the pathway activity in the same spot (intraview)
and the ten closest spots (paraview). Therefore we choose the family
`constant` and set l to ten, which will select the ten nearest
neighbors. Depending on the goal of the analysis, different families can
be applied.</p>
<p>We are also intrigued about the relationship of ligand expression and
pathway activity in the broader tissue. For this, we again create an
intra- and paraview, this time for the expression of the ligands, but
from this view, we only need the paraview. In the next step, we add it
to the pathway activity views to achieve our intended view
composition.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathway_act_view</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_initial_view.html">create_initial_view</a></span><span class="op">(</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">pathway_activity</span><span class="op">)</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/add_paraview.html">add_paraview</a></span><span class="op">(</span><span class="va">geometry</span>, l <span class="op">=</span> <span class="fl">10</span>, family <span class="op">=</span> <span class="st">"constant"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Generating paraview using 10 nearest neighbors per unit</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ligand_view</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_initial_view.html">create_initial_view</a></span><span class="op">(</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">ligand_expr</span><span class="op">)</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://sfirke.github.io/janitor/reference/clean_names.html" class="external-link">clean_names</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/add_paraview.html">add_paraview</a></span><span class="op">(</span><span class="va">geometry</span>, l <span class="op">=</span> <span class="fl">10</span>, family <span class="op">=</span> <span class="st">"constant"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Generating paraview using 10 nearest neighbors per unit</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">combined_views</span> <span class="op">&lt;-</span> <span class="va">pathway_act_view</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/add_views.html">add_views</a></span><span class="op">(</span><span class="fu"><a href="../reference/create_view.html">create_view</a></span><span class="op">(</span><span class="st">"paraview.ligand.10"</span>, <span class="va">ligand_view</span><span class="op">[[</span><span class="st">"paraview.10"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span>, <span class="st">"para.ligand.10"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Then run MISTy and collect the results:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/run_misty.html">run_misty</a></span><span class="op">(</span><span class="va">combined_views</span>, <span class="st">"result/functional_ligand"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/home/runner/work/mistyR/mistyR/vignettes/result/functional_ligand"</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/collect_results.html">collect_results</a></span><span class="op">(</span><span class="st">"result/functional_ligand/"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="downstream-analysis">Downstream analysis<a class="anchor" aria-label="anchor" href="#downstream-analysis"></a>
</h2>
<p>With the collected results, we can now answer the following
questions:</p>
<div class="section level3">
<h3 id="to-what-extent-can-the-analyzed-surrounding-tissues-activities-explain-the-pathway-activity-of-the-spot-compared-to-the-intraview">1. To what extent can the analyzed surrounding tissues’ activities
explain the pathway activity of the spot compared to the intraview?<a class="anchor" aria-label="anchor" href="#to-what-extent-can-the-analyzed-surrounding-tissues-activities-explain-the-pathway-activity-of-the-spot-compared-to-the-intraview"></a>
</h3>
<p>Here we can look at two different statistics: multi.R2 shows the
total variance explained by the multiview model. gain.R2 shows the
increase in explainable variance from the paraview.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_improvement_stats.html">plot_improvement_stats</a></span><span class="op">(</span><span class="st">"gain.R2"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_improvement_stats.html">plot_improvement_stats</a></span><span class="op">(</span><span class="st">"multi.R2"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 14 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_segment()`).</span></span></code></pre>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<pre><code><span><span class="co">## Warning: Removed 14 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_segment()`).</span></span></code></pre>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-9-2.png" width="700"></p>
<p>The paraview particularly increases the explained variance for TGFb
and PI3K. In general, the significant gain in R2 can be interpreted as
the following:</p>
<p>“We can better explain the expression of marker X when we consider
additional views other than the intrinsic view.”</p>
<p>To see the individual contributions of the views we can use:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_view_contributions.html">plot_view_contributions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="what-are-the-specific-relations-that-can-explain-the-pathway-activity">2. What are the specific relations that can explain the pathway
activity<a class="anchor" aria-label="anchor" href="#what-are-the-specific-relations-that-can-explain-the-pathway-activity"></a>
</h3>
<p>The importance of the markers from each viewpoint as predictors of
the spot intrinsic pathway activity can be shown individually to explain
the contributions.</p>
<p>First, for the intrinsic view. To set an importance threshold we
apply <code>cutoff</code>:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_interaction_heatmap.html">plot_interaction_heatmap</a></span><span class="op">(</span><span class="st">"intra"</span>, clean <span class="op">=</span> <span class="cn">TRUE</span>, cutoff <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>We can observe that TNFa is a significant predictor for the activity
of the NFkB pathway when in the same spot. Let’s take a look at the
spatial distribution of these pathway activities in the tissue
slide:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">seurat_vs</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tnfa"</span>, <span class="st">"nfkb"</span><span class="op">)</span>, image.alpha <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>We can observe a correlation between high TNFa activity and high NFkB
activity.</p>
<p>Now we repeat this analysis with the pathway activity paraview. With
<code>trim</code> we display only targets with a value above 0.5 for
<code>gain.R2</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_interaction_heatmap.html">plot_interaction_heatmap</a></span><span class="op">(</span>view <span class="op">=</span> <span class="st">"para.10"</span>, </span>
<span>                           clean <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                           trim <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                           trim.measure <span class="op">=</span> <span class="st">"gain.R2"</span>,</span>
<span>                           cutoff <span class="op">=</span> <span class="fl">1.25</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>From the <code>gain.R2</code> we know that the paraview contributes a
lot to explaining the TGFb pathway activity. Let’s visualize it and its
most important predictor, androgen pathway activity:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">seurat_vs</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"androgen"</span>, <span class="st">"tgfb"</span><span class="op">)</span>, image.alpha <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>The plots show us an anticorrelation of these pathways.</p>
<p>Now we will analyze the last view, the ligand expression
paraview:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_interaction_heatmap.html">plot_interaction_heatmap</a></span><span class="op">(</span>view <span class="op">=</span> <span class="st">"para.ligand.10"</span>, clean <span class="op">=</span> <span class="cn">TRUE</span>, trim <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                           trim.measure <span class="op">=</span> <span class="st">"gain.R2"</span>, cutoff<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>The ligand SERPINF1 is a predictor of both PI3K and VEGF:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">seurat_vs</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pi3k"</span><span class="op">)</span>, image.alpha <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">seurat_vs</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"vegf"</span><span class="op">)</span>, image.alpha <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-16-2.png" width="700"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_vs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"SCT"</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialFeaturePlot</a></span><span class="op">(</span><span class="va">seurat_vs</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"serpinf1"</span><span class="op">)</span>, image.alpha <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="FunctionalPipelinePathwayActivityLigands_files/figure-html/unnamed-chunk-16-3.png" width="700"></p>
<p>From the slides we can see that SERPINF1 correlates positive with
PI3K and negative with VEGF</p>
</div>
</div>
<div class="section level2">
<h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a>
</h2>
<p><code>browseVignettes("mistyR")</code></p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> at the point when
this document was compiled.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] distances_0.1.10   janitor_2.2.0      progeny_1.24.0     OmnipathR_3.10.1  </span></span>
<span><span class="co">##  [5] decoupleR_2.8.0    lubridate_1.9.3    forcats_1.0.0      stringr_1.5.1     </span></span>
<span><span class="co">##  [9] dplyr_1.1.4        purrr_1.0.2        readr_2.1.5        tidyr_1.3.1       </span></span>
<span><span class="co">## [13] tibble_3.2.1       ggplot2_3.5.0      tidyverse_2.0.0    Seurat_5.0.2      </span></span>
<span><span class="co">## [17] SeuratObject_5.0.1 sp_2.1-3           future_1.33.1      mistyR_1.10.0     </span></span>
<span><span class="co">## [21] BiocStyle_2.30.0  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RcppAnnoy_0.0.22       splines_4.3.3          later_1.3.2           </span></span>
<span><span class="co">##   [4] filelock_1.0.3         R.oo_1.26.0            cellranger_1.1.0      </span></span>
<span><span class="co">##   [7] polyclip_1.10-6        hardhat_1.3.1          pROC_1.18.5           </span></span>
<span><span class="co">##  [10] rpart_4.1.23           fastDummies_1.7.3      lifecycle_1.0.4       </span></span>
<span><span class="co">##  [13] globals_0.16.2         lattice_0.22-5         vroom_1.6.5           </span></span>
<span><span class="co">##  [16] MASS_7.3-60.0.1        backports_1.4.1        magrittr_2.0.3        </span></span>
<span><span class="co">##  [19] plotly_4.10.4          sass_0.4.8             rmarkdown_2.25        </span></span>
<span><span class="co">##  [22] jquerylib_0.1.4        yaml_2.3.8             rlist_0.4.6.2         </span></span>
<span><span class="co">##  [25] httpuv_1.6.14          sctransform_0.4.1      spam_2.10-0           </span></span>
<span><span class="co">##  [28] spatstat.sparse_3.0-3  reticulate_1.35.0      cowplot_1.1.3         </span></span>
<span><span class="co">##  [31] pbapply_1.7-2          RColorBrewer_1.1-3     abind_1.4-5           </span></span>
<span><span class="co">##  [34] rvest_1.0.4            Rtsne_0.17             R.utils_2.12.3        </span></span>
<span><span class="co">##  [37] nnet_7.3-19            rappdirs_0.3.3         ipred_0.9-14          </span></span>
<span><span class="co">##  [40] lava_1.7.3             ggrepel_0.9.5          irlba_2.3.5.1         </span></span>
<span><span class="co">##  [43] listenv_0.9.1          spatstat.utils_3.0-4   goftest_1.2-3         </span></span>
<span><span class="co">##  [46] RSpectra_0.16-1        spatstat.random_3.2-3  fitdistrplus_1.1-11   </span></span>
<span><span class="co">##  [49] parallelly_1.37.1      pkgdown_2.0.7          leiden_0.4.3.1        </span></span>
<span><span class="co">##  [52] codetools_0.2-19       xml2_1.3.6             tidyselect_1.2.0      </span></span>
<span><span class="co">##  [55] farver_2.1.1           stats4_4.3.3           matrixStats_1.2.0     </span></span>
<span><span class="co">##  [58] spatstat.explore_3.2-6 jsonlite_1.8.8         caret_6.0-94          </span></span>
<span><span class="co">##  [61] ellipsis_0.3.2         progressr_0.14.0       iterators_1.0.14      </span></span>
<span><span class="co">##  [64] ggridges_0.5.6         survival_3.5-8         systemfonts_1.0.5     </span></span>
<span><span class="co">##  [67] foreach_1.5.2          tools_4.3.3            progress_1.2.3        </span></span>
<span><span class="co">##  [70] ragg_1.2.7             ica_1.0-3              Rcpp_1.0.12           </span></span>
<span><span class="co">##  [73] glue_1.7.0             prodlim_2023.08.28     gridExtra_2.3         </span></span>
<span><span class="co">##  [76] ranger_0.16.0          xfun_0.42              withr_3.0.0           </span></span>
<span><span class="co">##  [79] BiocManager_1.30.22    fastmap_1.1.1          fansi_1.0.6           </span></span>
<span><span class="co">##  [82] digest_0.6.34          timechange_0.3.0       R6_2.5.1              </span></span>
<span><span class="co">##  [85] mime_0.12              textshaping_0.3.7      colorspace_2.1-0      </span></span>
<span><span class="co">##  [88] scattermore_1.2        tensor_1.5             spatstat.data_3.0-4   </span></span>
<span><span class="co">##  [91] R.methodsS3_1.8.2      utf8_1.2.4             generics_0.1.3        </span></span>
<span><span class="co">##  [94] recipes_1.0.10         data.table_1.15.2      class_7.3-22          </span></span>
<span><span class="co">##  [97] ridge_3.3              prettyunits_1.2.0      httr_1.4.7            </span></span>
<span><span class="co">## [100] htmlwidgets_1.6.4      ModelMetrics_1.2.2.2   uwot_0.1.16           </span></span>
<span><span class="co">## [103] pkgconfig_2.0.3        gtable_0.3.4           timeDate_4032.109     </span></span>
<span><span class="co">## [106] lmtest_0.9-40          selectr_0.4-2          furrr_0.3.1           </span></span>
<span><span class="co">## [109] htmltools_0.5.7        dotCall64_1.1-1        bookdown_0.38         </span></span>
<span><span class="co">## [112] scales_1.3.0           png_0.1-8              gower_1.0.1           </span></span>
<span><span class="co">## [115] snakecase_0.11.1       knitr_1.45             tzdb_0.4.0            </span></span>
<span><span class="co">## [118] reshape2_1.4.4         checkmate_2.3.1        nlme_3.1-164          </span></span>
<span><span class="co">## [121] curl_5.2.1             cachem_1.0.8           zoo_1.8-12            </span></span>
<span><span class="co">## [124] KernSmooth_2.23-22     parallel_4.3.3         miniUI_0.1.1.1        </span></span>
<span><span class="co">## [127] desc_1.4.3             pillar_1.9.0           grid_4.3.3            </span></span>
<span><span class="co">## [130] logger_0.2.2           vctrs_0.6.5            RANN_2.6.1            </span></span>
<span><span class="co">## [133] promises_1.2.1         xtable_1.8-4           cluster_2.1.6         </span></span>
<span><span class="co">## [136] evaluate_0.23          cli_3.6.2              compiler_4.3.3        </span></span>
<span><span class="co">## [139] rlang_1.1.3            crayon_1.5.2           future.apply_1.11.1   </span></span>
<span><span class="co">## [142] labeling_0.4.3         plyr_1.8.9             fs_1.6.3              </span></span>
<span><span class="co">## [145] stringi_1.8.3          viridisLite_0.4.2      deldir_2.0-4          </span></span>
<span><span class="co">## [148] assertthat_0.2.1       munsell_0.5.0          lazyeval_0.2.2        </span></span>
<span><span class="co">## [151] spatstat.geom_3.2-9    Matrix_1.6-5           RcppHNSW_0.6.0        </span></span>
<span><span class="co">## [154] hms_1.1.3              patchwork_1.2.0        bit64_4.0.5           </span></span>
<span><span class="co">## [157] shiny_1.8.0            highr_0.10             ROCR_1.0-11           </span></span>
<span><span class="co">## [160] igraph_2.0.2           memoise_2.0.1          bslib_0.6.1           </span></span>
<span><span class="co">## [163] bit_4.0.5              readxl_1.4.3</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jovan Tanevski.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
