<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="mistyR">
<title>Structural analysis with MISTy - based on DOT deconvolution • mistyR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Structural analysis with MISTy - based on DOT deconvolution">
<meta property="og:description" content="mistyR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-9"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-9');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mistyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.10.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/mistyR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/FunctionalAndStructuralPipeline.html">Learning functional and structural spatial relationships with MISTy</a>
    <a class="dropdown-item" href="../articles/FunctionalPipelinePathwayActivityLigands.html">Functional analysis with MISTy - pathway activity and ligand expression</a>
    <a class="dropdown-item" href="../articles/FunctionalPipelinePathwaySpecific.html">Functional analysis with MISTy - pathway specific genes</a>
    <a class="dropdown-item" href="../articles/MistyRStructuralAnalysisPipelineC2L.html">Structural analysis with MISTy - based on cell2location deconvolution</a>
    <a class="dropdown-item" href="../articles/MistyRStructuralAnalysisPipelineDOT.html">Structural analysis with MISTy - based on DOT deconvolution</a>
    <a class="dropdown-item" href="../articles/ReproduceSignaturePaper.html">MISTy representation based analysis of IMC breast cancer data</a>
    <a class="dropdown-item" href="../articles/mistyDataFormats.html">mistyR and data formats</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/mistyR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://saezlab.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Structural analysis with MISTy - based on DOT deconvolution</h1>
                        <h4 data-toc-skip class="author">Leoni
Zimmermann</h4>
            <address class="author_afil">
      Heidelberg University, Heidelberg,
Germany<br><h4 data-toc-skip class="author">Jovan
Tanevski</h4>
            <address class="author_afil">
      Heidelberg University and Heidelberg University Hospital,
Heidelberg, Germany <br>Jožef Stefan Institute, Ljubljana,
Slovenia<br><a class="author_email" href="mailto:#"></a><a href="mailto:jovan.tanevski@uni-heidelberg.de" class="email">jovan.tanevski@uni-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2024-03-20</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/mistyR/blob/HEAD/vignettes/MistyRStructuralAnalysisPipelineDOT.Rmd" class="external-link"><code>vignettes/MistyRStructuralAnalysisPipelineDOT.Rmd</code></a></small>
      <div class="d-none name"><code>MistyRStructuralAnalysisPipelineDOT.Rmd</code></div>
    </address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>MISTy is designed to analyze spatial omics datasets within and
between distinct spatial contexts referred to as views. This analysis
can focus solely on structural information. Spatial transcriptomic
methods such as Visium capture information from areas containing
multiple cells. Then, deconvolution is applied to relate the measured
data of the spots back to individual cells. In this vignette we will use
the R package <a href="https://saezlab.github.io/DOT/index.html" class="external-link"><code>DOT</code></a> for
deconvolution.</p>
<p>This vignette presents a workflow for the analysis of structural
data, guiding users through the application of <code>mistyR</code> to
the results of <code>DOT</code> deconvolution.</p>
<p>The package <code>DOT</code> can be installed from Github
<code>remotes::install_github("saezlab/DOT")</code>.</p>
<p>Load the necessary packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># MISTy </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/mistyR/">mistyR</a></span><span class="op">)</span> </span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://future.futureverse.org" class="external-link">future</a></span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># DOT</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/DOT/" class="external-link">DOT</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Loading experiment data </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mojaveazure.github.io/seurat-object/" class="external-link">SeuratObject</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Data manipulation </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Distances</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/fsavje/distances" class="external-link">distances</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-and-load-the-data">Get and load the data<a class="anchor" aria-label="anchor" href="#get-and-load-the-data"></a>
</h2>
<p>For this showcase, we use a 10X Visium spatial slide from <a href="https://doi.org/10.1038/s41586-022-05060-x" class="external-link">Kuppe et al.,
2022</a>, where they created a spatial multi-omic map of human
myocardial infarction. The tissue example data comes from the human
heart of patient 14 which is in a later state after myocardial
infarction. The Seurat object contains, among other things, the spot
coordinates on the slides which we will need for decomposition First, we
have to download and extract the file:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://zenodo.org/records/6580069/files/10X_Visium_ACH005.tar.gz?download=1"</span>,</span>
<span>    destfile <span class="op">=</span> <span class="st">"10X_Visium_ACH005.tar.gz"</span>, method <span class="op">=</span> <span class="st">"curl"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="st">"10X_Visium_ACH005.tar.gz"</span><span class="op">)</span></span></code></pre></div>
<p>The next step is to load the data and extract the location of the
spots. The rows are shifted, which means that the real distances between
two spots are not always the same. It is therefore advantageous to use
the pixel coordinates instead of row and column numbers, as the
distances between these are represented accurately.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spatial_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"ACH005/ACH005.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">geometry</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/GetTissueCoordinates.html" class="external-link">GetTissueCoordinates</a></span><span class="op">(</span><span class="va">spatial_data</span>, cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"imagerow"</span>, <span class="st">"imagecol"</span><span class="op">)</span>, scale <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p>For deconvolution, we additionally need a reference single-cell data
set containing a gene x cell count matrix and a vector containing the
corresponding cell annotations. Kuppe et al., 2022, obtained from each
sample isolated nuclei from the remaining tissue that they used for
snRNA-seq. The data corresponding to the same patient as the spatial
data will be used as reference data in <code>DOT</code>. First download
the file:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://www.dropbox.com/scl/fi/sq24xaavxplkc98iimvpz/hca_p14.rds?rlkey=h8cyxzhypavkydbv0z3pqadus&amp;dl=1"</span>,</span>
<span>              destfile <span class="op">=</span> <span class="st">"hca_p14.rds"</span>,</span>
<span>              mode <span class="op">=</span> <span class="st">"wb"</span><span class="op">)</span></span></code></pre></div>
<p>Now load the data. From this, we retrieve a gene x cell count matrix
and the respective cell annotations.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ref_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"hca_p14.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ref_counts_P14</span> <span class="op">&lt;-</span> <span class="va">ref_data</span><span class="op">$</span><span class="va">counts</span></span>
<span><span class="va">ref_ct</span> <span class="op">&lt;-</span> <span class="va">ref_data</span><span class="op">$</span><span class="va">celltypes</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="deconvolution-with-dot">Deconvolution with DOT<a class="anchor" aria-label="anchor" href="#deconvolution-with-dot"></a>
</h2>
<p>Next, we need to set up the DOT object. The two inputs we need are
the count matrix and pixel coordinates of the spatial data and the count
matrix and cell annotations of the single-cell reference data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dot.srt</span> <span class="op">&lt;-</span><span class="fu"><a href="https://saezlab.github.io/DOT/reference/setup.srt.html" class="external-link">setup.srt</a></span><span class="op">(</span>srt_data <span class="op">=</span> <span class="va">spatial_data</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">Spatial</span><span class="op">@</span><span class="va">counts</span>, srt_coords <span class="op">=</span> <span class="va">geometry</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">dot.ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/DOT/reference/setup.ref.html" class="external-link">setup.ref</a></span><span class="op">(</span>ref_data <span class="op">=</span> <span class="va">ref_counts_P14</span>, ref_annotations <span class="op">=</span> <span class="va">ref_ct</span>, <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/DOT/reference/create.DOT.html" class="external-link">create.DOT</a></span><span class="op">(</span><span class="va">dot.srt</span>, <span class="va">dot.ref</span><span class="op">)</span></span></code></pre></div>
<p>Now we can carry out deconvolution:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run DOT</span></span>
<span><span class="va">dot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/DOT/reference/run.DOT.lowresolution.html" class="external-link">run.DOT.lowresolution</a></span><span class="op">(</span><span class="va">dot</span><span class="op">)</span></span></code></pre></div>
<p>The results can be found under <code>dot@weights</code>. To obtain
the calculated cell-type proportion per spot, we normalize the result to
a row sum of 1.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Normalize DOT results</span></span>
<span><span class="va">DOT_weights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sweep.html" class="external-link">sweep</a></span><span class="op">(</span><span class="va">dot</span><span class="op">@</span><span class="va">weights</span>, <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">dot</span><span class="op">@</span><span class="va">weights</span><span class="op">)</span>, <span class="st">"/"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-cell-proportion-in-spots">Visualize cell proportion in spots<a class="anchor" aria-label="anchor" href="#visualize-cell-proportion-in-spots"></a>
</h2>
<p>Now we can visually explore the slide itself and the abundance of
cell types at each spot.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Tissue Slide</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/SpatialPlot.html" class="external-link">SpatialPlot</a></span><span class="op">(</span><span class="va">spatial_data</span>, keep.scale <span class="op">=</span> <span class="cn">NULL</span>, alpha <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> </span></code></pre></div>
<p><img src="MistyRStructuralAnalysisPipelineDOT_files/figure-html/unnamed-chunk-9-1.png" width="576"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Results DOT</span></span>
<span><span class="fu"><a href="https://saezlab.github.io/DOT/reference/draw_maps.html" class="external-link">draw_maps</a></span><span class="op">(</span><span class="va">geometry</span>, </span>
<span>          <span class="va">DOT_weights</span>, </span>
<span>          background <span class="op">=</span> <span class="st">"white"</span>, </span>
<span>          normalize <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>          ncol <span class="op">=</span> <span class="fl">3</span>, </span>
<span>          viridis_option <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MistyRStructuralAnalysisPipelineDOT_files/figure-html/unnamed-chunk-9-2.png" width="576"></p>
<p>Based on the plots, we can observe that some cell types are found
more frequently than others. Additionally, we can identify patterns in
the distribution of cells, with some being widespread across the entire
slide while others are concentrated in specific areas. Furthermore,
there are cell types that share a similar distribution.</p>
</div>
<div class="section level2">
<h2 id="misty-views">MISTy views<a class="anchor" aria-label="anchor" href="#misty-views"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculating the radius</span></span>
<span><span class="va">geom_dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/distances/man/distances.html" class="external-link">distances</a></span><span class="op">(</span><span class="va">geometry</span><span class="op">)</span><span class="op">)</span>  </span>
<span><span class="va">dist_nn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">geom_dist</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">paraview_radius</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">ceiling</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">dist_nn</span><span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html" class="external-link">sd</a></span><span class="op">(</span><span class="va">dist_nn</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create views</span></span>
<span><span class="va">heart_views</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_initial_view.html">create_initial_view</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">DOT_weights</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/add_paraview.html">add_paraview</a></span><span class="op">(</span><span class="va">geometry</span>, l<span class="op">=</span> <span class="va">paraview_radius</span>, family <span class="op">=</span> <span class="st">"gaussian"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run misty and collect results</span></span>
<span><span class="fu"><a href="../reference/run_misty.html">run_misty</a></span><span class="op">(</span><span class="va">heart_views</span>, <span class="st">"result/vignette_structural_pipeline"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/home/runner/work/mistyR/mistyR/vignettes/result/vignette_structural_pipeline"</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/collect_results.html">collect_results</a></span><span class="op">(</span><span class="st">"result/vignette_structural_pipeline"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="downstream-analysis">Downstream Analysis<a class="anchor" aria-label="anchor" href="#downstream-analysis"></a>
</h2>
<p>With the collected results, we can now answer the following
questions:</p>
<div class="section level4">
<h4 id="to-what-extent-can-the-occurring-cell-types-of-the-surrounding-tissue-explain-the-cell-type-composition-of-the-spot-compared-to-the-intraview">1. To what extent can the occurring cell types of the surrounding
tissue explain the cell type composition of the spot compared to the
intraview?<a class="anchor" aria-label="anchor" href="#to-what-extent-can-the-occurring-cell-types-of-the-surrounding-tissue-explain-the-cell-type-composition-of-the-spot-compared-to-the-intraview"></a>
</h4>
<p>Here we can look at two different statistics: <code>multi.R2</code>
shows the total variance explained by the multiview model.
<code>gain.R2</code> shows the increase in explainable variance from the
paraview.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_improvement_stats.html">plot_improvement_stats</a></span><span class="op">(</span><span class="st">"multi.R2"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/plot_improvement_stats.html">plot_improvement_stats</a></span><span class="op">(</span><span class="st">"gain.R2"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 11 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_segment()`).</span></span></code></pre>
<p><img src="MistyRStructuralAnalysisPipelineDOT_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<pre><code><span><span class="co">## Warning: Removed 11 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_segment()`).</span></span></code></pre>
<p><img src="MistyRStructuralAnalysisPipelineDOT_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
<p>The paraview particularly increases the explained variance for
adipocytes and mast cells. In general, the significant gain in
R<sup>2</sup> can be interpreted as the following:</p>
<p>“We can better explain the expression of marker X when we consider
additional views other than the intrinsic view.”</p>
</div>
<div class="section level4">
<h4 id="what-are-the-specific-relations-that-can-explain-the-contributions">2. What are the specific relations that can explain the
contributions?<a class="anchor" aria-label="anchor" href="#what-are-the-specific-relations-that-can-explain-the-contributions"></a>
</h4>
<p>To explain the contributions, we can visualize the importance of each
cell type in predicting the cell type distribution for each view
separately. With <code>trim</code>, we display only targets with a value
above 50 for <code>multi.R2</code>. To set an importance threshold we
would apply <code>cutoff</code>.</p>
<p>First, for the intrinsic view:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_interaction_heatmap.html">plot_interaction_heatmap</a></span><span class="op">(</span>view <span class="op">=</span> <span class="st">"intra"</span>, </span>
<span>                                           clean <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                           trim.measure <span class="op">=</span> <span class="st">"multi.R2"</span>,</span>
<span>                                           trim <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<p><img src="MistyRStructuralAnalysisPipelineDOT_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>We can observe that cardiomyocytes are a significant predictor for
some cell types when in the same spot. To identify the target with the
best prediction by cardiomyocytes, we can view the importance values as
follows:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span><span class="op">$</span><span class="va">importances.aggregated</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">view</span> <span class="op">==</span> <span class="st">"intra"</span>, <span class="va">Predictor</span> <span class="op">==</span> <span class="st">"CM"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="op">-</span><span class="va">Importance</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 11 × 5</span></span></span>
<span><span class="co">##    view  Predictor Target   Importance nsamples</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> intra CM        Fib           2.82         1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> intra CM        vSMCs         2.72         1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> intra CM        Endo          2.64         1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> intra CM        PC            2.60         1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> intra CM        Myeloid       2.42         1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> intra CM        Adipo         2.36         1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> intra CM        Mast          1.23         1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> intra CM        Lymphoid      0.685        1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> intra CM        prolif        0.461        1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> intra CM        Neuronal     -<span style="color: #BB0000;">0.117</span>        1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> intra CM        CM           <span style="color: #BB0000;">NA</span>            1</span></span></code></pre>
<p>Let’s take a look at the spatial distribution of Cardiomyocytes and
their most important target, fibroblasts, in the tissue slide:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://saezlab.github.io/DOT/reference/draw_maps.html" class="external-link">draw_maps</a></span><span class="op">(</span><span class="va">geometry</span>, </span>
<span>          <span class="va">DOT_weights</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Fib"</span>, <span class="st">"CM"</span><span class="op">)</span><span class="op">]</span>, </span>
<span>          background <span class="op">=</span> <span class="st">"white"</span>, </span>
<span>          size <span class="op">=</span> <span class="fl">1.25</span>, </span>
<span>          normalize <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>          ncol <span class="op">=</span> <span class="fl">1</span>,</span>
<span>          viridis_option <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MistyRStructuralAnalysisPipelineDOT_files/figure-html/unnamed-chunk-14-1.png" width="480"></p>
<p>We can observe that areas with high proportions of cardiomyocytes
have low proportions of fibroblasts and vice versa.</p>
<p>Now we repeat this analysis with the paraview:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_interaction_heatmap.html">plot_interaction_heatmap</a></span><span class="op">(</span>view <span class="op">=</span> <span class="st">"para.126"</span>, </span>
<span>                                           clean <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                           trim <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                                           trim.measure <span class="op">=</span> <span class="st">"gain.R2"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="MistyRStructuralAnalysisPipelineDOT_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>Here, we select the target adipocytes, as we know from previous
analysis that the paraview contributes a large part to explaining its
distribution. The best predictor for adipocytes are Myeloid cells. To
better identify the localization of the two cell types, we set the color
scaling to a smaller range, as there are a few spots with a high
proportion, which makes the distribution of spots with a low proportion
difficult to recognize.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://saezlab.github.io/DOT/reference/draw_maps.html" class="external-link">draw_maps</a></span><span class="op">(</span><span class="va">geometry</span>,</span>
<span>          <span class="va">DOT_weights</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Myeloid"</span>,<span class="st">"Adipo"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>          background <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>          size <span class="op">=</span> <span class="fl">1.25</span>,</span>
<span>          normalize <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>          ncol <span class="op">=</span> <span class="fl">1</span>,</span>
<span>          viridis_option <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_colour_viridis_c</a></span><span class="op">(</span>limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0.33</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="MistyRStructuralAnalysisPipelineDOT_files/figure-html/unnamed-chunk-16-1.png" width="480"></p>
<p>The plots show us that, in some places, the localization of the two
cell types overlap.</p>
</div>
</div>
<div class="section level2">
<h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a>
</h2>
<p><code>browseVignettes("mistyR")</code></p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> at the point when
this document was compiled.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.3 (2024-02-29)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] distances_0.1.10   lubridate_1.9.3    forcats_1.0.0      stringr_1.5.1     </span></span>
<span><span class="co">##  [5] dplyr_1.1.4        purrr_1.0.2        readr_2.1.5        tidyr_1.3.1       </span></span>
<span><span class="co">##  [9] tibble_3.2.1       ggplot2_3.5.0      tidyverse_2.0.0    Seurat_5.0.3      </span></span>
<span><span class="co">## [13] SeuratObject_5.0.1 sp_2.1-3           DOT_0.0.0.9000     future_1.33.1     </span></span>
<span><span class="co">## [17] mistyR_1.10.0      BiocStyle_2.30.0  </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RcppAnnoy_0.0.22       splines_4.3.3          later_1.3.2           </span></span>
<span><span class="co">##   [4] filelock_1.0.3         fields_15.2            R.oo_1.26.0           </span></span>
<span><span class="co">##   [7] polyclip_1.10-6        hardhat_1.3.1          pROC_1.18.5           </span></span>
<span><span class="co">##  [10] rpart_4.1.23           fastDummies_1.7.3      lifecycle_1.0.4       </span></span>
<span><span class="co">##  [13] vroom_1.6.5            globals_0.16.3         lattice_0.22-5        </span></span>
<span><span class="co">##  [16] MASS_7.3-60.0.1        magrittr_2.0.3         plotly_4.10.4         </span></span>
<span><span class="co">##  [19] sass_0.4.9             rmarkdown_2.26         jquerylib_0.1.4       </span></span>
<span><span class="co">##  [22] yaml_2.3.8             rlist_0.4.6.2          httpuv_1.6.14         </span></span>
<span><span class="co">##  [25] sctransform_0.4.1      spam_2.10-0            spatstat.sparse_3.0-3 </span></span>
<span><span class="co">##  [28] reticulate_1.35.0      cowplot_1.1.3          pbapply_1.7-2         </span></span>
<span><span class="co">##  [31] RColorBrewer_1.1-3     maps_3.4.2             abind_1.4-5           </span></span>
<span><span class="co">##  [34] Rtsne_0.17             R.utils_2.12.3         nnet_7.3-19           </span></span>
<span><span class="co">##  [37] ipred_0.9-14           lava_1.8.0             ggrepel_0.9.5         </span></span>
<span><span class="co">##  [40] irlba_2.3.5.1          listenv_0.9.1          spatstat.utils_3.0-4  </span></span>
<span><span class="co">##  [43] goftest_1.2-3          RSpectra_0.16-1        spatstat.random_3.2-3 </span></span>
<span><span class="co">##  [46] fitdistrplus_1.1-11    parallelly_1.37.1      pkgdown_2.0.7         </span></span>
<span><span class="co">##  [49] leiden_0.4.3.1         codetools_0.2-19       tidyselect_1.2.1      </span></span>
<span><span class="co">##  [52] farver_2.1.1           stats4_4.3.3           matrixStats_1.2.0     </span></span>
<span><span class="co">##  [55] spatstat.explore_3.2-6 jsonlite_1.8.8         caret_6.0-94          </span></span>
<span><span class="co">##  [58] ellipsis_0.3.2         progressr_0.14.0       ggridges_0.5.6        </span></span>
<span><span class="co">##  [61] survival_3.5-8         iterators_1.0.14       systemfonts_1.0.6     </span></span>
<span><span class="co">##  [64] foreach_1.5.2          tools_4.3.3            ragg_1.3.0            </span></span>
<span><span class="co">##  [67] ica_1.0-3              Rcpp_1.0.12            glue_1.7.0            </span></span>
<span><span class="co">##  [70] prodlim_2023.08.28     gridExtra_2.3          xfun_0.42             </span></span>
<span><span class="co">##  [73] ranger_0.16.0          withr_3.0.0            BiocManager_1.30.22   </span></span>
<span><span class="co">##  [76] fastmap_1.1.1          fansi_1.0.6            digest_0.6.35         </span></span>
<span><span class="co">##  [79] timechange_0.3.0       R6_2.5.1               mime_0.12             </span></span>
<span><span class="co">##  [82] textshaping_0.3.7      colorspace_2.1-0       scattermore_1.2       </span></span>
<span><span class="co">##  [85] tensor_1.5             spatstat.data_3.0-4    R.methodsS3_1.8.2     </span></span>
<span><span class="co">##  [88] utf8_1.2.4             generics_0.1.3         data.table_1.15.2     </span></span>
<span><span class="co">##  [91] recipes_1.0.10         class_7.3-22           httr_1.4.7            </span></span>
<span><span class="co">##  [94] ridge_3.3              htmlwidgets_1.6.4      ModelMetrics_1.2.2.2  </span></span>
<span><span class="co">##  [97] uwot_0.1.16            pkgconfig_2.0.3        gtable_0.3.4          </span></span>
<span><span class="co">## [100] timeDate_4032.109      lmtest_0.9-40          furrr_0.3.1           </span></span>
<span><span class="co">## [103] htmltools_0.5.7        dotCall64_1.1-1        bookdown_0.38         </span></span>
<span><span class="co">## [106] scales_1.3.0           png_0.1-8              gower_1.0.1           </span></span>
<span><span class="co">## [109] knitr_1.45             tzdb_0.4.0             reshape2_1.4.4        </span></span>
<span><span class="co">## [112] nlme_3.1-164           cachem_1.0.8           zoo_1.8-12            </span></span>
<span><span class="co">## [115] KernSmooth_2.23-22     parallel_4.3.3         miniUI_0.1.1.1        </span></span>
<span><span class="co">## [118] desc_1.4.3             pillar_1.9.0           grid_4.3.3            </span></span>
<span><span class="co">## [121] vctrs_0.6.5            RANN_2.6.1             promises_1.2.1        </span></span>
<span><span class="co">## [124] xtable_1.8-4           cluster_2.1.6          evaluate_0.23         </span></span>
<span><span class="co">## [127] cli_3.6.2              compiler_4.3.3         crayon_1.5.2          </span></span>
<span><span class="co">## [130] rlang_1.1.3            future.apply_1.11.1    labeling_0.4.3        </span></span>
<span><span class="co">## [133] plyr_1.8.9             fs_1.6.3               stringi_1.8.3         </span></span>
<span><span class="co">## [136] viridisLite_0.4.2      deldir_2.0-4           assertthat_0.2.1      </span></span>
<span><span class="co">## [139] munsell_0.5.0          lazyeval_0.2.2         spatstat.geom_3.2-9   </span></span>
<span><span class="co">## [142] Matrix_1.6-5           RcppHNSW_0.6.0         hms_1.1.3             </span></span>
<span><span class="co">## [145] patchwork_1.2.0        bit64_4.0.5            shiny_1.8.0           </span></span>
<span><span class="co">## [148] highr_0.10             ROCR_1.0-11            igraph_2.0.3          </span></span>
<span><span class="co">## [151] memoise_2.0.1          bslib_0.6.1            bit_4.0.5</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jovan Tanevski.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
