<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="mistyR">
<title>mistyR and data formats • mistyR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="mistyR and data formats">
<meta property="og:description" content="mistyR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-9"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-9');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mistyR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.10.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/mistyR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/mistyDataFormats.html">mistyR and data formats</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/mistyR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://saezlab.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>mistyR and data formats</h1>
                        <h4 data-toc-skip class="author">Ricardo Omar
Ramirez Flores</h4>
            <address class="author_afil">
      Heidelberg University and Heidelberg University Hospital,
Heidelberg,
Germany<br><h4 data-toc-skip class="author">Jovan
Tanevski</h4>
            <address class="author_afil">
      Heidelberg University and Heidelberg University Hospital,
Heidelberg, Germany <br>Jožef Stefan Institute, Ljubljana,
Slovenia<br><a class="author_email" href="mailto:#"></a><a href="mailto:jovan.tanevski@uni-heidelberg.de" class="email">jovan.tanevski@uni-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2023-12-22</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/saezlab/mistyR/blob/HEAD/vignettes/mistyDataFormats.Rmd" class="external-link"><code>vignettes/mistyDataFormats.Rmd</code></a></small>
      <div class="d-none name"><code>mistyDataFormats.Rmd</code></div>
    </address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.18/mistyR" class="external-link">mistyR</a></em> can
be used to analyze spatial omics data sets stored in <em><a href="https://bioconductor.org/packages/3.18/SpatialExperiment" class="external-link">SpatialExperiment</a></em>,
<em><a href="https://CRAN.R-project.org/package=SeuratObject" class="external-link">SeuratObject</a></em>
or <em><a href="https://CRAN.R-project.org/package=anndata" class="external-link">anndata</a></em>
object with just a couple of functions. In this vignette we demonstrate
how to build a user friendly workflow starting from data preprocessing,
through running <em><a href="https://bioconductor.org/packages/3.18/mistyR" class="external-link">mistyR</a></em>, to
analysis of results, focusing on functional representation of 10x Visium
data.</p>
<p>First load packages needed for the analysis.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># MISTy</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/mistyR/">mistyR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://future.futureverse.org" class="external-link">future</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># data manipulation</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># normalization</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/satijalab/sctransform" class="external-link">sctransform</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># resource</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/decoupleR/" class="external-link">decoupleR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plotting</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># setup parallel execution</span></span>
<span><span class="fu"><a href="https://future.futureverse.org/reference/plan.html" class="external-link">plan</a></span><span class="op">(</span><span class="va">multisession</span><span class="op">)</span></span></code></pre></div>
<p>As an example, we will analyze a 10X Visium spatial gene expression
dataset of one breast cancer section (Block A Section 1) available here
[<a href="https://support.10xgenomics.com/spatial-gene-expression/datasets" class="external-link uri">https://support.10xgenomics.com/spatial-gene-expression/datasets</a>].
For convenience, we make this dataset available as object in several
popular formats including <em><a href="https://bioconductor.org/packages/3.18/SpatialExperiment" class="external-link">SpatialExperiment</a></em>,
<em><a href="https://CRAN.R-project.org/package=SeuratObject" class="external-link">SeuratObject</a></em>
or <em><a href="https://CRAN.R-project.org/package=anndata" class="external-link">anndata</a></em> and
demonstrate how to extract the data needed for further analysis from
these objects.</p>
<p>We will explore the spatial interactions of the Hypoxia pathway
responsive genes with the Estrogen pathway responsive genes. To this end
we will convert the count based representation of the data to a
representation based on pathway specific genes and/or estimated pathway
activities. To this end we will use the package <em><a href="https://bioconductor.org/packages/3.18/decoupleR" class="external-link">decoupleR</a></em>.</p>
</div>
<div class="section level2">
<h2 id="get-the-data">Get the data<a class="anchor" aria-label="anchor" href="#get-the-data"></a>
</h2>
<div class="section level3">
<h3 id="spatialexperiment">SpatialExperiment<a class="anchor" aria-label="anchor" href="#spatialexperiment"></a>
</h3>
<p>To load the packages needed to work with <em><a href="https://bioconductor.org/packages/3.18/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
data and download the object.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># SpatialExperiment</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: SingleCellExperiment</span></span>
<span><span class="co">#&gt; Loading required package: SummarizedExperiment</span></span>
<span><span class="co">#&gt; Loading required package: MatrixGenerics</span></span>
<span><span class="co">#&gt; Loading required package: matrixStats</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'matrixStats'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     count</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'MatrixGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">#&gt;     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">#&gt;     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">#&gt;     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">#&gt;     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">#&gt;     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">#&gt;     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">#&gt;     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">#&gt;     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">#&gt;     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">#&gt;     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">#&gt;     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">#&gt;     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">#&gt;     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">#&gt;     rowWeightedSds, rowWeightedVars</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     combine, intersect, setdiff, union</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     first, rename</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:Matrix':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand, unname</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'IRanges'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:purrr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     reduce</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:dplyr':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     collapse, desc, slice</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biobase'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     rowMedians</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyMissing, rowMedians</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://www.dropbox.com/scl/fi/7mdmz6vk10ib55qn7w3fw/visium_spe.rds?rlkey=j03qamdm9zcin577chlabms0m&amp;dl=1"</span>,</span>
<span>    destfile <span class="op">=</span> <span class="st">"visium_spe.rds"</span>, mode <span class="op">=</span><span class="st">"wb"</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Next, read the object and extract the expression and location
data.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe.vs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"visium_spe.rds"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Expression data</span></span>
<span><span class="va">expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">spe.vs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># deal with duplicate names</span></span>
<span><span class="va">symbols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">spe.vs</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span></span>
<span><span class="va">d.index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">symbols</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">symbols</span><span class="op">[</span><span class="va">d.index</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">symbols</span><span class="op">[</span><span class="va">d.index</span><span class="op">]</span>,<span class="st">".1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">symbols</span></span>
<span></span>
<span><span class="co"># Location data</span></span>
<span><span class="va">geometry</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">spe.vs</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">array_row</span>, <span class="va">array_col</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geometry</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"row"</span>, <span class="st">"col"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="seurat">Seurat<a class="anchor" aria-label="anchor" href="#seurat"></a>
</h3>
<p>To load the packages needed to work with <em><a href="https://CRAN.R-project.org/package=SeuratObject" class="external-link">SeuratObject</a></em>
data and download the object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Seurat</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://www.dropbox.com/scl/fi/44zf4le1xcq7ichjp11bg/visium_seurat.rds?rlkey=ikrrsp2rncqde0nnsbdte1joa&amp;dl=1"</span>,</span>
<span>    destfile <span class="op">=</span> <span class="st">"visium_seurat.rds"</span>, mode <span class="op">=</span><span class="st">"wb"</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Next, read the object and extract the expression and location
data.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat.vs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"visium_seurat.rds"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Expression data</span></span>
<span><span class="va">expression</span> <span class="op">&lt;-</span> <span class="fu">GetAssayData</span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">seurat.vs</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"Spatial"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Seurat deals with duplicates internally in similar way as above</span></span>
<span></span>
<span><span class="co"># Location data</span></span>
<span><span class="va">geometry</span> <span class="op">&lt;-</span> <span class="fu">GetTissueCoordinates</span><span class="op">(</span><span class="va">seurat.vs</span>,</span>
<span>    cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"row"</span>, <span class="st">"col"</span><span class="op">)</span>, scale <span class="op">=</span> <span class="cn">NULL</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="anndata">AnnData<a class="anchor" aria-label="anchor" href="#anndata"></a>
</h3>
<p>To load the packages needed to work with <em><a href="https://CRAN.R-project.org/package=SeuratObject" class="external-link">SeuratObject</a></em>
data and download the object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># AnnData</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://anndata.dynverse.org" class="external-link">anndata</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"https://www.dropbox.com/scl/fi/jubijl0pr8rhka8mfjpcn/visium_anndata.h5ad?rlkey=xmhmfl5oz61dgngmackkklp32&amp;dl=1"</span>,</span>
<span>    destfile <span class="op">=</span> <span class="st">"visium_anndata.h5ad"</span>, mode <span class="op">=</span><span class="st">"wb"</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Next, read the object and extract the expression and location
data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anndata.vs</span> <span class="op">&lt;-</span> <span class="fu">read_h5ad</span><span class="op">(</span><span class="st">"visium_anndata.h5ad"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Expression data</span></span>
<span><span class="co"># Here the dgRMatrix is converted to a dense matrix for vst compatibility reasons</span></span>
<span><span class="va">expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">anndata.vs</span><span class="op">$</span><span class="va">X</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># deal with duplicate names (alternatively see AnnData.var_names_make_unique)</span></span>
<span><span class="va">symbols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span></span>
<span><span class="va">d.index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">symbols</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">symbols</span><span class="op">[</span><span class="va">d.index</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">symbols</span><span class="op">[</span><span class="va">d.index</span><span class="op">]</span>,<span class="st">".1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">expression</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">symbols</span></span>
<span></span>
<span><span class="co"># Location data</span></span>
<span><span class="va">geometry</span> <span class="op">&lt;-</span> <span class="va">anndata.vs</span><span class="op">$</span><span class="va">obs</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"array_row"</span>, <span class="st">"array_col"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geometry</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"row"</span>, <span class="st">"col"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="preprocess-the-data">Preprocess the data<a class="anchor" aria-label="anchor" href="#preprocess-the-data"></a>
</h2>
<p>From here on, the analysis proceeds in the same way no matter the
input format.</p>
<div class="section level3">
<h3 id="normalize-counts-and-drop-duplicates">Normalize counts and drop duplicates<a class="anchor" aria-label="anchor" href="#normalize-counts-and-drop-duplicates"></a>
</h3>
<p>In this example we normalize the counts using vst normalization.
However, the user must define what’s the best solution for their
analysis.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">norm.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/sctransform/man/vst.html" class="external-link">vst</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">expression</span>, <span class="st">"dgCMatrix"</span><span class="op">)</span>, verbosity <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">$</span><span class="va">y</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filtering-genes-that-are-expressed-in-at-least-5-of-the-spots">Filtering genes that are expressed in at least 5% of the spots<a class="anchor" aria-label="anchor" href="#filtering-genes-that-are-expressed-in-at-least-5-of-the-spots"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coverage</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">norm.data</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">norm.data</span><span class="op">)</span></span>
<span><span class="va">slide.markers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">coverage</span><span class="op">[</span><span class="va">coverage</span> <span class="op">&gt;=</span> <span class="fl">0.05</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="run-misty-on-pathway-specific-genes">Run MISTy on pathway specific genes<a class="anchor" aria-label="anchor" href="#run-misty-on-pathway-specific-genes"></a>
</h2>
<p>In this use casewe would like to dissect the relationships between
Estrogen and Hypoxia responsive genes coming from two spatial contexts:
Relationships within the spot and relationships in the broader tissue
structure. In particular, as intrinsic representation we will use the
normalized counts of Hypoxia responsive genes. We will explore the
relationships between the Hypoxia responsive genes at the context of a
Visium spot, the relationships between Hypoxia responsive genes in the
broader tissue structure and the relationships between the Estrogen
responsive genes and Hypoxia responsive genes in the broader tissue
structure.</p>
<div class="section level3">
<h3 id="defining-estrogen-and-hypoxia-responsive-genes">Defining Estrogen and Hypoxia responsive genes<a class="anchor" aria-label="anchor" href="#defining-estrogen-and-hypoxia-responsive-genes"></a>
</h3>
<p>For this simple example we will pick the top 15 most significantly
responsive genes of each pathway from the model matrix from the resource
<code>progeny</code> available from the package <em><a href="https://bioconductor.org/packages/3.18/decoupleR" class="external-link">decoupleR</a></em>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">resource</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://saezlab.github.io/decoupleR/reference/get_progeny.html" class="external-link">get_progeny</a></span><span class="op">(</span>organism <span class="op">=</span><span class="st">"human"</span>, top <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: One or more parsing issues, call `problems()` on your data frame for details,</span></span>
<span><span class="co">#&gt; e.g.:</span></span>
<span><span class="co">#&gt;   dat &lt;- vroom(...)</span></span>
<span><span class="co">#&gt;   problems(dat)</span></span>
<span></span>
<span><span class="va">estrogen.footprints</span> <span class="op">&lt;-</span> <span class="va">resource</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">source</span> <span class="op">==</span> <span class="st">"Estrogen"</span>, <span class="va">weight</span> <span class="op">!=</span> <span class="fl">0</span>, <span class="va">target</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">slide.markers</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">target</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hypoxia.footprints</span> <span class="op">&lt;-</span> <span class="va">resource</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">source</span> <span class="op">==</span> <span class="st">"Hypoxia"</span>, <span class="va">weight</span> <span class="op">!=</span> <span class="fl">0</span>, <span class="va">target</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">slide.markers</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">target</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="view-composition">View composition<a class="anchor" aria-label="anchor" href="#view-composition"></a>
</h3>
<p>To capture the relationships of interest within the descrbed
contexts, our MISTy model will consist of three views. First we
construct a Hypoxia specific intraview (capturing the expression within
a spot) and add a Hypoxia specific paraview (capturing the expression in
the broader tissue structure) with a significance radius of 5 spots.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hypoxia.views</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_initial_view.html">create_initial_view</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">norm.data</span><span class="op">)</span><span class="op">[</span>, <span class="va">hypoxia.footprints</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/add_paraview.html">add_paraview</a></span><span class="op">(</span><span class="va">geometry</span>, l<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Generating paraview</span></span></code></pre></div>
<p>We will next create a similar view composition but using Estrogen
reponsive genes. This is an easy way to generate a view that will
capture the expression of Estrogen responsive genes in the broader
tissue structure that we will add to the previously generated view
composition.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">estrogen.views</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_initial_view.html">create_initial_view</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">norm.data</span><span class="op">)</span><span class="op">[</span>,<span class="va">estrogen.footprints</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/add_paraview.html">add_paraview</a></span><span class="op">(</span><span class="va">geometry</span>, l<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Generating paraview</span></span></code></pre></div>
<p>We next combine the view composition in a composition capturing all
relationships of interest: Hypoxia intraview + Hypoxia paraview +
Estrogen paraview.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty.views</span> <span class="op">&lt;-</span> <span class="va">hypoxia.views</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/add_views.html">add_views</a></span><span class="op">(</span><span class="fu"><a href="../reference/create_view.html">create_view</a></span><span class="op">(</span><span class="st">"paraview.estrogen.5"</span>, <span class="va">estrogen.views</span><span class="op">[[</span><span class="st">"paraview.5"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span>, <span class="st">"para.estrogen.5"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-misty-and-collect-results">Run MISTy and collect results<a class="anchor" aria-label="anchor" href="#run-misty-and-collect-results"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/run_misty.html">run_misty</a></span><span class="op">(</span><span class="va">misty.views</span>, <span class="st">"vignette_model_footprints"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/home/runner/work/mistyR/mistyR/vignettes/vignette_model_footprints"</span></span>
<span></span>
<span><span class="va">misty.results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/collect_results.html">collect_results</a></span><span class="op">(</span><span class="st">"vignette_model_footprints"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="interpretation-and-downstream-analysis">Interpretation and downstream analysis<a class="anchor" aria-label="anchor" href="#interpretation-and-downstream-analysis"></a>
</h3>
<p>MISTy gives explanatory answers to three general questions:</p>
<p><strong>1. How much can the broader spatial context explain the
expression of markers (in contrast to the intraview)?</strong></p>
<p>This can be observed in the gain in R2 (or RMSE) of using the
multiview model in contrast to the single <code>main</code> view
model.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty.results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_improvement_stats.html">plot_improvement_stats</a></span><span class="op">(</span><span class="st">"gain.R2"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/plot_improvement_stats.html">plot_improvement_stats</a></span><span class="op">(</span><span class="st">"gain.RMSE"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-15-1.png" width="700"><img src="mistyDataFormats_files/figure-html/unnamed-chunk-15-2.png" width="700"></p>
<p>In this example, PGK1 is a marker whose expression can be explained
better by modeling the broader spatial context around each spot.</p>
<p>We can further inspect the significance of the gain in variance
explained, by the assigned p-value of improvement based on
cross-validation.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty.results</span><span class="op">$</span><span class="va">improvements</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">measure</span> <span class="op">==</span> <span class="st">"p.R2"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">value</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 4</span></span></span>
<span><span class="co">#&gt;    target  sample                                                measure   value</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PGK1    /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    3.32<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> ANKZF1  /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    1.15<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> INSIG2  /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    2.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PDK1    /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    1.72<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> FAM162A /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    3.34<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> FUT11   /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    4.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> EGLN1   /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    6.05<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> HILPDA  /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    6.11<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> NDRG1   /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    6.81<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> ENO2    /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    9.40<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> PFKFB4  /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    1.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> KDM3A   /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    1.30<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> BNIP3L  /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    1.53<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> ANKRD37 /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    3.34<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> GBE1    /home/runner/work/mistyR/mistyR/vignettes/vignette_m… p.R2    7.74<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span></span></span></code></pre></div>
<p>In general, the significant gain in R2 can be interpreted as the
following:</p>
<p>“We can better explain the expression of marker X, when we consider
additional views, other than the intrinsic view.”</p>
<p><strong>2.How much do different view components contribute to
explaining the expression?</strong></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty.results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_view_contributions.html">plot_view_contributions</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">misty.results</span><span class="op">$</span><span class="va">contributions.stats</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">target</span> <span class="op">==</span> <span class="st">"PGK1"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 6</span></span></span>
<span><span class="co">#&gt;   target view             mean fraction p.mean  p.sd</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PGK1   intra           0.517    0.375      0    <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> PGK1   para.5          0.356    0.258      0    <span style="color: #BB0000;">NA</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PGK1   para.estrogen.5 0.508    0.368      0    <span style="color: #BB0000;">NA</span></span></span></code></pre></div>
<p>In the case of PGK1, we observe that around 37.7% of the contribution
in the final model comes from the expression of other markers of hypoxia
intrinsically or from the broader tissue structure. The rest (62.3%)
comes from the expression of estrogen and hypoxia responsive genes from
the broader tissue structure.</p>
<p><strong>3.What are the specific relations that can explain the
contributions?</strong></p>
<p>To explain the contributions, we can visualize the importances of
markers coming from each view separately as predictors of the expression
of the intrinsic markers of hypoxia.</p>
<p>First, the intrinsic importances of the hypoxia markers.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty.results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_interaction_heatmap.html">plot_interaction_heatmap</a></span><span class="op">(</span>view <span class="op">=</span> <span class="st">"intra"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<p>These importances are associated to the relationship between markers
in the same spot. Let’s pick the best predictor of PGK1 to confirm
this:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty.results</span><span class="op">$</span><span class="va">importances.aggregated</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">view</span> <span class="op">==</span> <span class="st">"intra"</span>, <span class="va">Target</span> <span class="op">==</span> <span class="st">"PGK1"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="op">-</span><span class="va">Importance</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 5</span></span></span>
<span><span class="co">#&gt;    view  Predictor Target Importance nsamples</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> intra NDRG1     PGK1       2.46          1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> intra FAM162A   PGK1       1.97          1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> intra PFKFB4    PGK1       0.291         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> intra HILPDA    PGK1       0.277         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> intra ENO2      PGK1      -<span style="color: #BB0000;">0.062</span><span style="color: #BB0000; text-decoration: underline;">0</span>        1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> intra EGLN1     PGK1      -<span style="color: #BB0000;">0.284</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> intra GBE1      PGK1      -<span style="color: #BB0000;">0.415</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> intra KDM3A     PGK1      -<span style="color: #BB0000;">0.462</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> intra BNIP3L    PGK1      -<span style="color: #BB0000;">0.562</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> intra ANKRD37   PGK1      -<span style="color: #BB0000;">0.587</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> intra INSIG2    PGK1      -<span style="color: #BB0000;">0.592</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> intra ANKZF1    PGK1      -<span style="color: #BB0000;">0.597</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> intra FUT11     PGK1      -<span style="color: #BB0000;">0.711</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> intra PDK1      PGK1      -<span style="color: #BB0000;">0.733</span>         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> intra PGK1      PGK1      <span style="color: #BB0000;">NA</span>             1</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">vis.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">geometry</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">norm.data</span><span class="op">)</span><span class="op">[</span>,<span class="va">hypoxia.footprints</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">norm.data</span><span class="op">)</span><span class="op">[</span>,<span class="va">estrogen.footprints</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">vis.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">col</span>, y<span class="op">=</span><span class="va">row</span>, color <span class="op">=</span> <span class="va">PGK1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_void</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">vis.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">col</span>, y<span class="op">=</span><span class="va">row</span>, color <span class="op">=</span> <span class="va">NDRG1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_void</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-20-2.png" width="700"></p>
<p>Second, the paraview importances of the hypoxia markers.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty.results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_interaction_heatmap.html">plot_interaction_heatmap</a></span><span class="op">(</span>view <span class="op">=</span> <span class="st">"para.5"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
<p>These importances are associated to the relationship between markers
in the spot and markers in the neighborhood (controlled by our parameter
l).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">vis.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">col</span>, y<span class="op">=</span><span class="va">row</span>, color <span class="op">=</span> <span class="va">PGK1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_void</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">vis.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">col</span>, y<span class="op">=</span><span class="va">row</span>, color <span class="op">=</span> <span class="va">EGLN1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_void</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-23-2.png" width="700"></p>
<p>As expected, the expression of EGLN1 (a predictor with hign
importance from this view) in the neighborhood of each spot allows to
explain the expression of PGK1.</p>
<p>Finally, the paraview importances of the estrogen markers. We will
inspect the best predictor in this view.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">misty.results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_interaction_heatmap.html">plot_interaction_heatmap</a></span><span class="op">(</span>view <span class="op">=</span> <span class="st">"para.estrogen.5"</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">vis.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">col</span>, y<span class="op">=</span><span class="va">row</span>, color <span class="op">=</span> <span class="va">PGK1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_void</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-25-1.png" width="700"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">vis.data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">col</span>, y<span class="op">=</span><span class="va">row</span>, color <span class="op">=</span> <span class="va">TPD52L1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_void</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="mistyDataFormats_files/figure-html/unnamed-chunk-25-2.png" width="700"></p>
<p>It is visible that in some areas the local expression of TPD52L1
overlaps with the areas with the highest expression of PGK1.</p>
</div>
<div class="section level3">
<h3 id="important-notes">Important notes<a class="anchor" aria-label="anchor" href="#important-notes"></a>
</h3>
<ul>
<li><p>The relationships captured in the importances are not to assumed
or interpreted as linear or casual.</p></li>
<li><p>1-to-1 importances between predictor and markers should always be
interpreted in the context of the other predictors, since training MISTy
models is multivariate predictive task.</p></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="other-use-cases">Other use cases<a class="anchor" aria-label="anchor" href="#other-use-cases"></a>
</h2>
<p>The shown example is not the only way to use mistyR to analyze
spatial transcriptomics data. Similar and complementary workflows can be
constructed to describe different aspects of biology, for example:</p>
<ul>
<li><p>Spatial interactions between pathway activities and putative
ligands, as shown <a href="https://doi.org/10.1101/2020.05.08.084145" class="external-link">here</a>.</p></li>
<li><p>Spatial interactions between cell-state lineage markers and
putative ligands, as shown <a href="https://doi.org/10.1101/2020.12.08.411686" class="external-link">here</a>.</p></li>
<li><p>Spatial interactions between cell-type abundances leveraging
deconvolution methods and creating descriptions of cell colocalization
and tissue architecture.</p></li>
</ul>
<p>Additionally, <em><a href="https://github.com/saezlab/mistyR" class="external-link">mistyR</a></em> through the
function <code><a href="../reference/collect_results.html">collect_results()</a></code> allows you to group the results
of multiple slides, allowing for a more robust, integrative or
comparative analysis of spatial interactions.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a>
</h2>
<div class="section level3">
<h3 class="unnumbered" id="more-examples">More examples<a class="anchor" aria-label="anchor" href="#more-examples"></a>
</h3>
<p><code>browseVignettes("mistyR")</code></p>
<p><a href="https://saezlab.github.io/mistyR/articles/">Online
articles</a></p>
</div>
<div class="section level3">
<h3 class="unnumbered" id="publication">Publication<a class="anchor" aria-label="anchor" href="#publication"></a>
</h3>
<p><em>Jovan Tanevski, Ricardo Omar Ramirez Flores, Attila Gabor, Denis
Schapiro, Julio Saez-Rodriguez. Explainable multiview framework for
dissecting spatial relationships from highly multiplexed data. Genome
Biology 23, 97 (2022). <a href="https://doi.org/10.1186/s13059-022-02663-5" class="external-link uri">https://doi.org/10.1186/s13059-022-02663-5</a></em></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> at the point when
this document was compiled:</p>
<pre><code><span><span class="co">#&gt; R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] SpatialExperiment_1.12.0    SingleCellExperiment_1.24.0</span></span>
<span><span class="co">#&gt;  [3] SummarizedExperiment_1.32.0 Biobase_2.62.0             </span></span>
<span><span class="co">#&gt;  [5] GenomicRanges_1.54.1        GenomeInfoDb_1.38.2        </span></span>
<span><span class="co">#&gt;  [7] IRanges_2.36.0              S4Vectors_0.40.2           </span></span>
<span><span class="co">#&gt;  [9] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      </span></span>
<span><span class="co">#&gt; [11] matrixStats_1.2.0           ggplot2_3.4.4              </span></span>
<span><span class="co">#&gt; [13] decoupleR_2.8.0             sctransform_0.4.1          </span></span>
<span><span class="co">#&gt; [15] purrr_1.0.2                 dplyr_1.1.4                </span></span>
<span><span class="co">#&gt; [17] tibble_3.2.1                Matrix_1.6-1.1             </span></span>
<span><span class="co">#&gt; [19] future_1.33.1               mistyR_1.10.0              </span></span>
<span><span class="co">#&gt; [21] BiocStyle_2.30.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3      jsonlite_1.8.8          magrittr_2.0.3         </span></span>
<span><span class="co">#&gt;   [4] magick_2.8.2            farver_2.1.1            rmarkdown_2.25         </span></span>
<span><span class="co">#&gt;   [7] fs_1.6.3                zlibbioc_1.48.0         ragg_1.2.7             </span></span>
<span><span class="co">#&gt;  [10] vctrs_0.6.5             memoise_2.0.1           RCurl_1.98-1.13        </span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.7         S4Arrays_1.2.0          progress_1.2.3         </span></span>
<span><span class="co">#&gt;  [16] curl_5.2.0              cellranger_1.1.0        SparseArray_1.2.2      </span></span>
<span><span class="co">#&gt;  [19] sass_0.4.8              parallelly_1.36.0       bslib_0.6.1            </span></span>
<span><span class="co">#&gt;  [22] desc_1.4.3              plyr_1.8.9              lubridate_1.9.3        </span></span>
<span><span class="co">#&gt;  [25] cachem_1.0.8            igraph_1.6.0            lifecycle_1.0.4        </span></span>
<span><span class="co">#&gt;  [28] pkgconfig_2.0.3         R6_2.5.1                fastmap_1.1.1          </span></span>
<span><span class="co">#&gt;  [31] GenomeInfoDbData_1.2.11 digest_0.6.33           selectr_0.4-2          </span></span>
<span><span class="co">#&gt;  [34] colorspace_2.1-0        furrr_0.3.1             textshaping_0.3.7      </span></span>
<span><span class="co">#&gt;  [37] filelock_1.0.3          labeling_0.4.3          fansi_1.0.6            </span></span>
<span><span class="co">#&gt;  [40] timechange_0.2.0        httr_1.4.7              abind_1.4-5            </span></span>
<span><span class="co">#&gt;  [43] compiler_4.3.2          bit64_4.0.5             withr_2.5.2            </span></span>
<span><span class="co">#&gt;  [46] backports_1.4.1         logger_0.2.2            OmnipathR_3.10.1       </span></span>
<span><span class="co">#&gt;  [49] highr_0.10              R.utils_2.12.3          MASS_7.3-60            </span></span>
<span><span class="co">#&gt;  [52] rappdirs_0.3.3          DelayedArray_0.28.0     rjson_0.2.21           </span></span>
<span><span class="co">#&gt;  [55] tools_4.3.2             future.apply_1.11.1     R.oo_1.25.0            </span></span>
<span><span class="co">#&gt;  [58] glue_1.6.2              grid_4.3.2              checkmate_2.3.1        </span></span>
<span><span class="co">#&gt;  [61] reshape2_1.4.4          generics_0.1.3          gtable_0.3.4           </span></span>
<span><span class="co">#&gt;  [64] tzdb_0.4.0              R.methodsS3_1.8.2       tidyr_1.3.0            </span></span>
<span><span class="co">#&gt;  [67] data.table_1.14.10      hms_1.1.3               xml2_1.3.6             </span></span>
<span><span class="co">#&gt;  [70] utf8_1.2.4              XVector_0.42.0          pillar_1.9.0           </span></span>
<span><span class="co">#&gt;  [73] stringr_1.5.1           vroom_1.6.5             later_1.3.2            </span></span>
<span><span class="co">#&gt;  [76] lattice_0.21-9          bit_4.0.5               tidyselect_1.2.0       </span></span>
<span><span class="co">#&gt;  [79] knitr_1.45              gridExtra_2.3           bookdown_0.37          </span></span>
<span><span class="co">#&gt;  [82] xfun_0.41               stringi_1.8.3           yaml_2.3.8             </span></span>
<span><span class="co">#&gt;  [85] evaluate_0.23           codetools_0.2-19        BiocManager_1.30.22    </span></span>
<span><span class="co">#&gt;  [88] cli_3.6.2               systemfonts_1.0.5       munsell_0.5.0          </span></span>
<span><span class="co">#&gt;  [91] jquerylib_0.1.4         Rcpp_1.0.11             readxl_1.4.3           </span></span>
<span><span class="co">#&gt;  [94] globals_0.16.2          parallel_4.3.2          pkgdown_2.0.7          </span></span>
<span><span class="co">#&gt;  [97] readr_2.1.4             assertthat_0.2.1        prettyunits_1.2.0      </span></span>
<span><span class="co">#&gt; [100] bitops_1.0-7            listenv_0.9.0           rlist_0.4.6.2          </span></span>
<span><span class="co">#&gt; [103] scales_1.3.0            crayon_1.5.2            rlang_1.1.2            </span></span>
<span><span class="co">#&gt; [106] rvest_1.0.3             distances_0.1.10</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jovan Tanevski.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
